packages/block-editor/src/hooks/test/use-typography-props.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Internal dependencies
+ */
+import { getTypographyClassesAndStyles } from '../use-typography-props';
+
+describe( 'getTypographyClassesAndStyles', () => {
+	it( 'should return styles and classes', () => {
+		const attributes = {
+			fontFamily: 'tofu',
+			fontSize: 'large',
+			style: {
+				typography: {
+					letterSpacing: '22px',
+					fontSize: '2rem',
+					fontFamily: 'tahini',
+				},
+			},
+		};
+		expect( getTypographyClassesAndStyles( attributes ) ).toEqual( {
+			className: 'has-tofu-font-family has-large-font-size',
+			style: {
+				letterSpacing: '22px',
+				fontSize: '2rem',
+				fontFamily: 'tahini',
```

Comment:
Not sure I get this. So we could have an element with a `has-tofu-font-family` classname and an actual font-family of `tahini`? Shouldn't the classname be generated from the style?

Reply:
The font-family class is generated from the `fontFamily` block attribute. That's done independently from whatever is in the style attribute. The inline styles, however, are generated solely from the style attribute. My assumption of @ramonjd's intent here was that it checked they didn't impact each other.

In terms of how the block supports themselves function, I don't think we can have a block with a named font family (i.e. `fontFamily` attribute set) as well as an inline style for one (`style.typography.fontFamily`).

Reply:
Yeah, I'd expect that the presence of an inline style _would_ impact the classnames because they shouldn't both coexist if that makes sense? Though I get this function doesn't touch that logic, as documentation the test feels confusing. Not a blocker though 😅

Reply:
I agree on both counts. 

You could also argue that the presence of an inline style for our block supports does impact the classname as when it is set the top-level `fontFamily` block attribute gets cleared, effectively clearing the classname.

I'll replace the `fontFamily` inline style with something else to avoid further confusion.

Reply:
Tweaked the test via https://github.com/WordPress/gutenberg/pull/43499/commits/d0aa4a3cb6ebee75c3ec83cf2320f9e68bd2048e